### PR TITLE
Dark mode on web

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Enable dark mode on web for users that are part of a work-in-progress test that [requires opting in](https://www.theguardian.com/opt/in/dark-mode-web).

## Why?

Following from #11328 we discovered up to a third of page views have their colour scheme preference set to **dark**.

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before][] | ![after][] |

[before]: https://github.com/guardian/dotcom-rendering/assets/76776/ffa93420-1a83-4569-9067-0277fa36531b
[after]: https://github.com/guardian/dotcom-rendering/assets/76776/3b07269b-1150-4852-be4d-bc05f3970ec3

### CODE REVIEW

File: dotcom-rendering/src/components/ArticlePage.tsx [hunk 1]
```diff
@@ -45,19 +45,15 @@ interface AppProps extends BaseProps {
...
 	:root {
 		/* Light palette is default on all platforms */
 		${paletteDeclarations(format, 'light')}
 		body {
 			color: ${sourcePalette.neutral[7]};
 		}
-		/* Dark palette only for apps and only if switch turned on */
-		${darkModeInApps && renderingTarget === 'Apps'
+		/* Dark palette only if supported */
+		${darkMode
```
Comment (Jakeii) [inline dotcom-rendering/src/components/ArticlePage.tsx:56]: enableDarkMode
```suggestion
		${enableDarkMode
```

File: dotcom-rendering/src/components/ArticlePage.tsx [hunk 2]
```diff
@@ -45,19 +45,15 @@ interface AppProps extends BaseProps {
  * @description
  * returns global styles for article pages
  */
-const globalStyles = (
-	format: ArticleFormat,
-	darkModeInApps: boolean,
-	renderingTarget: RenderingTarget,
-) => css`
+const globalStyles = (format: ArticleFormat, darkMode: boolean) => css`
```
Comment (Jakeii) [inline dotcom-rendering/src/components/ArticlePage.tsx:48]:
```suggestion
const globalStyles = (format: ArticleFormat, enableDarkMode: boolean) => css`
```

File: dotcom-rendering/src/components/ArticlePage.tsx [hunk 3]
```diff
@@ -97,16 +93,13 @@ export const ArticlePage = (props: WebProps | AppProps) => {
 
 	const isWeb = renderingTarget === 'Web';
 	const webLightbox = isWeb && !!article.config.switches.lightbox;
+	const darkMode = isWeb
```
Comment (Jakeii) [inline dotcom-rendering/src/components/ArticlePage.tsx:96]:
```suggestion
	const enableDarkMode = isWeb
```

File: dotcom-rendering/src/components/ArticlePage.tsx [hunk 4]
```diff
@@ -97,16 +93,13 @@ export const ArticlePage = (props: WebProps | AppProps) => {
...
 	return (
 		<StrictMode>
-			<Global
-				styles={globalStyles(
-					format,
-					!!article.config.switches.darkModeInApps,
-					renderingTarget,
-				)}
-			/>
+			<Global styles={globalStyles(format, darkMode)} />
```
Comment (Jakeii) [inline dotcom-rendering/src/components/ArticlePage.tsx:102]:
```suggestion
			<Global styles={globalStyles(format, enableDarkMode)} />
```

File: dotcom-rendering/src/server/dev-index.html [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<!doctype html>
+i<!doctype html>
```
Comment (Jakeii) [inline dotcom-rendering/src/server/dev-index.html:1]:
```suggestion
<!doctype html>
```

Comment (mxdvl) [reply dotcom-rendering/src/server/dev-index.html:1]: how funny!

File: dotcom-rendering/src/components/ArticlePage.tsx (hunk 1 above)
Comment (Jakeii) [inline dotcom-rendering/src/components/ArticlePage.tsx:56]: Better to keep consistent and use `darkModeAvailable` everywhere?

**Review by Jakeii — APPROVED** [review]
Lovely 🌚

File: dotcom-rendering/src/components/ArticlePage.tsx (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/components/ArticlePage.tsx:56]: Not sure I understand…?

Comment (Jakeii) [reply dotcom-rendering/src/components/ArticlePage.tsx:56]: yeah `darkModeAvailable` is used down here https://github.com/guardian/dotcom-rendering/pull/11341/files#diff-a11eccfa997d0d1e670d52c7a3fefeeb296a34ec6411b5e8870aa66b56e69dccR54 seems more semantic?

**Comment by mxdvl** [comment]
> Nice touch with the dev index page, how do the `Link` and `ButtonFace` background colours work?

They are [System colours, part of CSS colour level 4](https://developer.mozilla.org/en-US/docs/Web/CSS/system-color)

**Pull request merged**